build-tools/packages/build-cli/src/library/repoPolicyCheck/npmPackages.ts

```diff
@@ -505,6 +509,211 @@ function getPreferredScriptLine(scriptLine: string): string {
+							}
+						})
+						.catch(() =>
+							console.warn(
+								`Error parsing API Extractor config: ${configFileAbsPath} for ${packageJson.name} "${name}".`,
+							),
+						),
+				);
+			}
+		}
+	}
+	await Promise.all(promises);
+}
+
+interface ScriptEntry {
+	name: string;
+	commandLine: string;
+}
+
+/**
+ * Determine missing elements to properly lint API exports.
+ *
+ * @remarks Does not check that api-extractor config files have linting enabled.
+ *
+ * @param packageJson - package.json contents
+ * @param dir - directory of package.json
+ * @returns record of missing requirements or with unexpected values
+ */
+async function getApiLintElementsMissing(
+	packageJson: Readonly<PackageJson>,
+	dir: string,
+): Promise<{
+	scriptEntries: ScriptEntry[];
+	configFiles: Map<string, string>;
+	devDependencies: string[];
+	targetsImpacted: Set<string>;
+}> {
+	const scriptEntries: ScriptEntry[] = [];
+	const configFiles = new Map<string, string>();
+	const devDependencies: string[] = [];
+	const targetsImpacted = new Set<string>();
+	const missing = { scriptEntries, configFiles, devDependencies, targetsImpacted };
+
+	const exportsField = packageJson.exports;
+	if (exportsField === undefined) {
+		return missing;
+	}
+
+	const { mapTypesPathToExportPaths } = queryTypesResolutionPathsFromPackageExports(
+		packageJson,
+		new Map([[/\.d\.ts$/, undefined]]),
+		{ node10TypeCompat: false, onlyFirstMatches: false },
+	);
+
+	const needsLinted = new Map<string, string>();
+	let bundleLintTarget: string | undefined;
+	for (const [relPath, exports] of mapTypesPathToExportPaths.entries()) {
+		const onlyRequire = exports.every((e) => e.conditions.includes("require"));
+		const onlyImport = exports.every((e) => e.conditions.includes("import"));
+		const skew = onlyRequire ? "cjs:" : onlyImport ? "esm:" : "";
+		if (exports.some((e) => !e.exportPath.startsWith("./internal"))) {
+			const existingSkew = needsLinted.get(relPath);
+			if (existingSkew === undefined) {
+				needsLinted.set(relPath, skew);
+			} else if (existingSkew !== skew) {
+				needsLinted.set(relPath, "");
+			}
+		} else if (exports.some((e) => e.exportPath === "./internal")) {
+			// ./internal export should be checked for cross repo consistency.
+			// Only one file needs to be checked for this. Prefer export that
+			// is not 'require' restricted.
+			// eslint-disable-next-line unicorn/no-lonely-if
+			if (bundleLintTarget === undefined || !onlyRequire) {
+				bundleLintTarget = relPath;
+			}
+		}
+	}
+	if (needsLinted.size === 0) {
+		// No files need linting or the only file that should be linted is internal,
+		// which is not checked by policy
+		return missing;
+	}
+
+	// -------------------------------------------------------------------------
+	// There are files that need linting.
+
+	// Make sure the package.json has the check:exports script that runs others.
+	const checkExports = packageJson.scripts?.["check:exports"];
```

Comment:
There should be a policy already that lets us configure required scripts. I'd rather make several smaller policies that check only one or two things and combine them rather than one policy that checks a lot of different things.

Reply:
I'll go with the collective wisdom.
I view this all as one policy that exports are linted. There are multiple pieces to that policy. I don't like it if there is one thing that has to be coordinated in several places without a clear thing tying them together.